src/components/ProgressBar/src/ProgressBar.vue

```diff
@@ -0,0 +1,113 @@
+<template>
+	<div
+		:class="[
+			$s.ProgressBarContainer,
+			$s[`size_${size}`],
+			$s[`shape_${shape}`],
+		]"
+	>
+		<div
+			:class="[
+				$s.ProgressBar,
+				$s[`shape_${shape}`],
+			]"
+			:style="{
+				backgroundColor: color,
+				transform: `translateX(-${progressTranslate}%)`
+			}"
+		/>
+	</div>
+</template>
+
+<script>
+import chroma from 'chroma-js';
+
+const MIN_PROGRESS = 0;
+const MAX_PROGRESS = 100;
+
+export default {
+	name: 'ProgressBar',
+
+	props: {
+		/**
+		 * Size (height) of the progress bar
+		 */
+		size: {
+			type: String,
+			default: 'medium',
+			validator: (size) => ['xsmall', 'small', 'medium', 'large'].includes(size),
+		},
+		/**
+		 * Shape of the progress bar
+		 */
+		shape: {
+			type: String,
+			default: 'rounded',
+			validator: (shape) => ['squared', 'rounded', 'pill'].includes(shape),
+		},
+		/**
+		 * Color of the progress bar (not background)
+		 */
+		color: {
+			type: String,
+			default: '#000',
+			validator: (color) => chroma.valid(color),
+		},
+		/**
+		 * Progress/width of the bar (0-100)
+		 */
+		progress: {
+			type: Number,
+			default: 0,
+			validator: (progress) => progress >= MIN_PROGRESS && progress <= MAX_PROGRESS,
+		},
+	},
+
+	computed: {
+		progressTranslate() {
+			return MAX_PROGRESS - this.progress;
+		},
+	},
+};
+</script>
+
+<style module="$s">
+.ProgressBarContainer {
+	width: 100%;
+	overflow: hidden;
+	background-color: #f5f4f4;
+}
+
+.ProgressBar {
+	height: 100%;
+	transition: transform 100ms linear;
```

Comment:
I think a similar thing can be done `transition: width ...` and it's less expensive than transforms. See orbit's implementation for ideas: https://github.com/square/orbit/blob/master/src/components/ProgressBar/src/ProgressBar.vue